src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -935,6 +1034,403 @@ public override Task DelayAsync(DelayAsyncInput input)
                     }
                 });
             }
+
+            /// <inheritdoc />
+            public override Task<TResult> ScheduleActivityAsync<TResult>(
+                ScheduleActivityInput input)
+            {
+                if (input.Options.StartToCloseTimeout == null &&
+                    input.Options.ScheduleToCloseTimeout == null)
+                {
+                    throw new ArgumentException("Activity options must have StartToCloseTimeout or ScheduleToCloseTimeout");
+                }
+
+                return ExecuteActivityInternalAsync<TResult>(
+                    local: false,
+                    doBackoff =>
```

Comment:
Backoffs cannot apply to non-local activities. You should just error here or log an error or whatever instead of having all this code.

I was tempted to create two kinds of resolution proto messages to make this unrepresentable, but that was a bit overkill.

Reply:
Yeah, I can error if I ever get called with a non-null backoff. Here I was just ignoring it knowing that core will never call this a second time so the enforcement didn't seem necessary, but i can add it.